plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/fileio/ForwardingFileIo.java

```diff
@@ -88,6 +92,36 @@ public void deleteFiles(Iterable<String> pathsToDelete)
         partitions.forEach(this::deleteBatch);
     }
 
+    @Override
+    public InputFile newInputFile(ManifestFile manifest)
+    {
+        checkArgument(
+                manifest.keyMetadata() == null,
+                "Cannot decrypt manifest: %s (use EncryptingFileIO)",
+                manifest.path());
+        return newInputFile(manifest.path(), manifest.length());
+    }
+
+    @Override
+    public InputFile newInputFile(DataFile file)
+    {
+        checkArgument(
+                file.keyMetadata() == null,
+                "Cannot decrypt data file: %s (use EncryptingFileIO)",
+                file.path());
+        return newInputFile(file.path().toString(), file.fileSizeInBytes());
+    }
+
+    @Override
+    public InputFile newInputFile(DeleteFile file)
+    {
+        checkArgument(
+                file.keyMetadata() == null,
+                "Cannot decrypt delete file: %s (use EncryptingFileIO)",
+                file.path());
+        return newInputFile(file.path().toString(), file.fileSizeInBytes());
```

Comment:
Does https://github.com/apache/iceberg/pull/9953 obsolete this?

at the minimum, let's have a TODO comment with a link

Reply:
added